gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/CopySource.java

```diff
@@ -291,6 +292,20 @@ private void submitUnfulfilledRequestEventsHelper(List<FileSet<CopyEntity>> file
     }
   }
 
+  void failJobIfAllRequestsRejected(RequestAllocator<FileSet<CopyEntity>> allocator,
+      Iterator<FileSet<CopyEntity>> allocatedRequests) throws IOException {
+    // TODO: we should set job as partial success if there is a mix of allocated requests and rejections
```

Comment:
I think this partial success is seen frequently as well, worth to address.

Reply:
I realize that we don't have a job status specifically for partial success, it ends up being marked as a total success unless there is a partial job commit policy. Should we change this to be if the job policy is marked as partial success, then it would pass?

We only have the following states so far:
```
		PREPARING, RUNNING, INTERRUPTED, CANCELLED, SUCCESS, FAILED
```

So adding a new state would be part of a larger change

Reply:
Yeah, no need to handle in this PR, but it worth to think how can we not silent fail some datasets when the resources is not enough and at the same time we want other dataset been processed successfully.